### PR TITLE
move references to sub-folders into workflow files

### PR DESCRIPTION
**Pull request by wxtim**
Rationalize where constants are kept. Came out of reviewing #5140 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests not required because there has been not functional change

### CODE REVIEW

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -193,6 +193,12 @@ class WorkflowFiles:
     LOG_DIR = 'log'
     """Workflow log directory."""
 
+    LOG_INSTALL_DIR = os.path.join(LOG_DIR, 'install')
+    """Install log dir"""
+
+    LOG_VERSION_DIR = os.path.join(LOG_DIR, 'version')
+    """Version control log dir"""
+
```
Comment (MetRonnie) [inline cylc/flow/workflow_files.py:201]: Would it make sense to move these to a subclass `Log` á la `Service` and `Install` below?

**Review by MetRonnie — APPROVED** [review]

File: tests/unit/post_install/test_log_vc_info.py [hunk 2]
```diff
@@ -167,7 +166,8 @@ def test_write_diff_git(git_source_repo: Tuple[str, str], tmp_path: Path):
     """Test write_diff() for a git repo"""
     source_dir, _ = git_source_repo
     run_dir = tmp_path / 'run_dir'
-    (run_dir / WorkflowFiles.LOG_DIR).mkdir(parents=True)
+    breakpoint()
```
Comment (datamel) [inline tests/unit/post_install/test_log_vc_info.py:169]: Breakpoint needs removing.
```suggestion
```

**Review by datamel — COMMENTED** [review]
Looks good, just debug removing needed from me.

**Review by MetRonnie — COMMENTED** [review]
Got some functional test failures

File: cylc/flow/install_plugins/log_vc_info.py [hunk 3]
```diff
@@ -265,7 +263,8 @@ def write_vc_info(
     """
     if not info:
         raise ValueError("Nothing to write")
-    info_file = Path(run_dir, LOG_VERSION_DIR, INFO_FILENAME)
+    info_file = Path(
+        run_dir, WorkflowFiles.LogDir.VERSION, INFO_FILENAME)
```
Comment (datamel) [inline cylc/flow/install_plugins/log_vc_info.py:267]: This isn't the same as the original path. We need `run_dir/log/version`, at the moment it is making `run_dir/version`.

**Review by datamel — CHANGES_REQUESTED** [review]

File: cylc/flow/install_plugins/log_vc_info.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/install_plugins/log_vc_info.py]: Thanks Mel - this has turned out to be a bigger refactor than intended

**Comment by datamel** [comment]
Looks ready, is re-review needed @wxtim?

**Review by datamel — APPROVED** [review]
👍

**Pull request merged**